src/gwt/src/org/rstudio/studio/client/projects/ui/prefs/buildtools/ProjectBuildToolsPreferencesPane.java

```diff
@@ -109,11 +109,21 @@ public RestartRequirement onApply(RProjectOptions options)
       for (BuildToolsPanel panel : buildToolsPanels_.values())
          panel.save(options);
 
-      // require reload if the build type or roxygen settings changed
+      boolean reload = false;
+
+      // reload if the build type or roxygen settings changed
       String initialBuildType = initialConfig_.getBuildType();
       String selectedBuildType = buildTypeSelect_.getValue();
+      if (initialBuildType != selectedBuildType)
+         reload = true;
+
+      // reload if "clean before install" changed
+      boolean initialClean = initialConfig_.getPackageCleanBeforeInstall();
+      boolean selectedClean = config.getPackageCleanBeforeInstall();
+      if (initialClean != selectedClean)
+         reload = true;
```

Comment:
This may be a bit strong, instead of this I'd just want to be able to call the `BuildPane::updateTooltips()` method.

Reply:
Or maybe updating/having the tooltip is not worth the trouble ? and a generic tooltip mentioning that the button can be configured with the "configure build tools... " is enough ?

Reply:
It's probably okay to leave this as-is; it's unlikely users will change this very often (especially if we're exposing separate buttons for clean and "incremental" installs)